driver-legacy/src/main/com/mongodb/Mongo.java

```diff
@@ -326,6 +322,7 @@ public Mongo(
         this.readConcern = options.getReadConcern() != null ? options.getReadConcern() : ReadConcern.DEFAULT;
         this.optionHolder = new Bytes.OptionHolder(null);
         this.credentialsList = unmodifiableList(credentialsList);
+        this.delegate = new MongoClientDelegate(cluster, credentialsList, this);
```

Comment:
Added MongoClientDelegate to share logic between this class and the new com.mongodb.client.MongoClientImpl.